pkg/minikube/driver/driver.go

```diff
@@ -141,17 +145,69 @@ func IsDocker(name string) bool {
 	return name == Docker
 }
 
-// IsDockerDesktop checks if the driver is a Docker for Desktop (Docker on windows or MacOs)
+func IsLinuxDockerDesktop(name string) bool {
+	if IsDocker(name) {
+		cmd := exec.Command("docker", "version")
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			return false
+		}
+		if err := cmd.Start(); err != nil {
+			return false
+		}
+		data, err := ioutil.ReadAll(stdout)
+		if err != nil {
+			return false
+		}
+		output := string(data)
+		return strings.Contains(output, "Desktop")
+	}
+	return false
+}
+
+// IsDockerDesktop checks if the driver is a Docker for Desktop
 // for linux and exotic archs, this will be false
 func IsDockerDesktop(name string) bool {
 	if IsDocker(name) {
 		if runtime.GOOS == "darwin" || runtime.GOOS == "windows" {
 			return true
 		}
+		return IsLinuxDockerDesktop(name)
 	}
 	return false
 }
 
+// GenerateClientForLinuxDockerDesktop returns the host of docker daemon when using linux docker desktop
+func GenerateClientForLinuxDockerDesktop() (*client.Client, error) {
+	cmd := exec.Command("docker", "context", "inspect", "desktop-linux")
+	stdout, _ := cmd.StdoutPipe()
+	cmd.Start()
+	data, _ := ioutil.ReadAll(stdout)
+	var objs []interface{}
+	json.Unmarshal(data, &objs)
+	objForDesktopLinux, ok := objs[0].(map[string]interface{})
+	if !ok {
+		cli, _ := client.NewClientWithOpts(client.FromEnv)
+		return cli, fmt.Errorf("failed to run acquire uinx sock host for docker desktop")
+	}
+	objForEndpoints, ok := objForDesktopLinux["Endpoints"].(map[string]interface{})
+	if !ok {
+		cli, _ := client.NewClientWithOpts(client.FromEnv)
+		return cli, fmt.Errorf("failed to run acquire uinx sock host for docker desktop")
+	}
+	objForDocker, ok := objForEndpoints["docker"].(map[string]interface{})
+	if !ok {
+		cli, _ := client.NewClientWithOpts(client.FromEnv)
+		return cli, fmt.Errorf("failed to run acquire uinx sock host for docker desktop")
+	}
+	objforHost, ok := objForDocker["Host"].(string)
+	if !ok {
+		cli, _ := client.NewClientWithOpts(client.FromEnv)
+		return cli, fmt.Errorf("failed to run acquire uinx sock host for docker desktop")
+	}
```

Comment:
Doing type casts for each is hard to read and makes it hard if we want to check for new values in the future. It would be better to use a struct with JSON tags.
```
type obj struct {
    Endpoints struct {
        Docker struct {
            Host string `json:"Host"`
        } `json:"docker"`
    } `json:"Endpoints"`
}
```